source/DSCResources/DSC_CMCollections/DSC_CMCollections.psm1

```diff
@@ -0,0 +1,685 @@
+    }
+    finally
+    {
+        Set-Location -Path "$env:temp"
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will test the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER CollectionName
+        Specifies a name for the collection.
+
+    .PARAMETER CollectionType
+        Specifies the type of collection. Valid values are User and Device.
+
+    .PARAMETER LimitingCollectionName
+        Specifies the name of a collection to use as the default scope for this collection.
+        Limiting collection is not evaluated in Test and is only used
+        if the collection needs created.
+
+    .PARAMETER Comment
+        Specifies a comment for the collection.
+
+    .PARAMETER RefreshSchedule
+        Specifies a schedule that determines when Configuration Manager refreshes the collection.
+
+    .PARAMETER RefreshType
+        Specifies how Configuration Manager refreshes the collection.
+        Valid values are: Manual, Periodic, Continuous, and Both.
+
+    .PARAMETER QueryRules
+        Specifies the name of the Rule and the query expression that Configuration Manager uses to update collections.
+
+    .PARAMETER ExcludeMembership
+        Specifies the collection name to exclude members from. If clients are in the excluded collection they will
+        not be added to the collection.
+
+    .PARAMETER DirectMembership
+        Specifies the resourceid for the direct membership rule.
+
+    .PARAMETER Ensure
+        Specifies if the collection is to be present or absent.
+#>
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $CollectionName,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('User','Device')]
+        [String]
+        $CollectionType,
+
+        [Parameter()]
+        [String]
+        $LimitingCollectionName,
+
+        [Parameter()]
+        [String]
+        $Comment,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $RefreshSchedule,
+
+        [Parameter()]
+        [ValidateSet('None','Periodic','Continuous','Both')]
+        [String]
+        $RefreshType,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $QueryRules,
+
+        [Parameter()]
+        [String[]]
+        $ExcludeMembership,
+
+        [Parameter()]
+        [String[]]
+        $DirectMembership,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -CollectionName $CollectionName -CollectionType $CollectionType
+    $cn = Get-CMCollection -Name $CollectionName
```

Comment:
If there is something else in Get that will make it know that the Collection doesn't exist this wouldn't need to be called. Or you could change all the state to cn instead.